pop-api/examples/dao/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use ink::{
+	prelude::{string::String, vec::Vec},
+	storage::Mapping,
+};
+use pop_api::{
+	primitives::TokenId,
+	v0::fungibles::{
+		self as api,
+		events::{Approval, Created, Transfer},
+		Psp22Error,
+	},
+};
+
+pub const STRINGLIMIT: u8 = u8::MAX;
+#[cfg(test)]
+mod tests;
+
+#[ink::contract]
+mod dao {
+	use super::*;
+
+	/// Structure of the proposal used by the Dao governance sysytem
+	#[derive(Debug, Clone)]
+	#[ink::scale_derive(Encode, Decode, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(ink::storage::traits::StorageLayout))]
+	pub struct Proposal {
+		// Description of the proposal
+		pub description: Vec<u8>,
+
+		// Beginnning of the voting period for this proposal
+		pub vote_start: BlockNumber,
+
+		// End of the voting period for this proposal
+		pub vote_end: BlockNumber,
+
+		// Balance representing the total votes for this proposal
+		pub yes_votes: Balance,
+
+		// Balance representing the total votes against this proposal
+		pub no_votes: Balance,
+
+		// Flag that indicates if the proposal was executed
+		pub executed: bool,
+
+		// AccountId of the recipient of the proposal
+		pub beneficiary: AccountId,
+
+		// Amount of tokens to be awarded to the beneficiary
+		pub amount: Balance,
+
+		// Identifier of the proposal
+		pub proposal_id: u32,
+	}
+
+	/// Representation of a member in the voting system
+	#[derive(Debug)]
+	#[ink::scale_derive(Encode, Decode, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(ink::storage::traits::StorageLayout))]
+	pub struct Member {
+		// Stores the member's voting influence by using his balance
+		pub voting_power: Balance,
+
+		// Keeps track of the last vote casted by the member
+		pub last_vote: BlockNumber,
+	}
+
+	/// Structure of a DAO (Decentralized Autonomous Organization)
+	/// that uses Psp22 to manage the Dao treasury and funds projects
+	/// selected by the members through governance
+	#[ink(storage)]
+	pub struct Dao {
+		// Funding proposals
+		proposals: Mapping<u32, Proposal>,
+
+		// Mapping of AccountId to Member structs, representing DAO membership.
+		members: Mapping<AccountId, Member>,
+
+		// Mapping tracking the last time each account voted.
+		last_votes: Mapping<AccountId, BlockNumber>,
```

Comment:
I don't see this field is used anywhere. Why do we need it?

Reply:
Oups! forgot to update this field in the `vote` function, corrected!